cosmwasm/deploy-contract.js

```diff
@@ -0,0 +1,404 @@
+        }
+
+        case 'VotingVerifier': {
+            if (!chain) {
+                throw new Error('VotingVerifier requires chainNames option');
+            }
+
+            return makeVotingVerifierInstantiateMsg(contractConfig, contracts, chain);
+        }
+
+        case 'Gateway': {
+            if (!chain) {
+                throw new Error('Gateway requires chainNames option');
+            }
+
+            return makeGatewayInstantiateMsg(contracts, chain);
+        }
+
+        case 'MultisigProver': {
+            if (!chain) {
+                throw new Error('MultisigProver requires chainNames option');
+            }
+
+            return makeMultisigProverInstantiateMsg(contractConfig, contracts, chain);
+        }
+    }
+
+    throw new Error(`${contractName} is not supported.`);
+};
+
+const deploy = async (options, chain, config) => {
+    printInfo('Deploying for chain', chain ? chain.name : 'none');
+
+    const wallet = await DirectSecp256k1HdWallet.fromMnemonic(options.mnemonic, { prefix: 'axelar' });
+    const client = await SigningCosmWasmClient.connectWithSigner(config.axelar.rpc, wallet);
+
+    if (config.axelar.contracts[options.contractName] === undefined) {
+        config.axelar.contracts[options.contractName] = {};
+    }
+
+    const contractConfig = config.axelar.contracts[options.contractName];
+
+    printInfo('Contract name', options.contractName);
+
+    const reuseCodeId = !!options.reuseCodeId && !!contractConfig.codeId;
+    printInfo('Reusing codeId', reuseCodeId.toString());
+
+    if (!reuseCodeId) {
+        const result = await uploadContract(config, options, wallet, client);
+        contractConfig.codeId = result.codeId;
+
+        if (result.address) {
+            contractConfig.address = result.address;
+            printInfo('Expected contract address', contractConfig.address);
+        }
+    }
+
+    printInfo('Code Id', contractConfig.codeId);
+
+    if (!options.uploadOnly) {
+        const initMsg = makeInstantiateMsg(options.contractName, config, chain);
+        const contractAddress = await instantiateContract(config, options, options.contractName, initMsg, wallet, client);
+
+        if (chain) {
+            contractConfig[chain.id] = {
+                ...contractConfig[chain.id],
+                address: contractAddress,
+            };
+        } else {
+            contractConfig.address = contractAddress;
+        }
+
+        printInfo('Contract address', contractAddress);
+    }
+};
+
+const main = async (options) => {
+    const config = loadConfig(options.env);
+
+    let chains = options.chainNames.split(',').map((str) => str.trim());
+
+    if (options.chainNames === 'all') {
+        chains = Object.keys(config.chains);
+    }
+
+    for (const chain of chains) {
+        if (config.chains[chain.toLowerCase()] === undefined && chain !== 'none') {
+            throw new Error(`Chain ${chain} is not defined in the info file`);
+        }
+    }
```

Comment:
```suggestion
    const undefinedChain = chains.find((chain) => !config.chains[chain.toLowerCase()] && chain !== 'none');

    if (undefinedChain) {
        throw new Error(`Chain ${undefinedChain} is not defined in the info file`);
    }
```
js is actually a very functional language.